File: Easy/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        const int n = nums.size();
        int last = n;
        for (int i = n - 1; i >= 0; i--) {
            if (nums[i] == val) {
                swap(nums[i], nums[--last]);
            }
        }
        return last;
    }
};

File: Algorithms/Array/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        count = len(nums)
        i = 0
        while i < count:
            if nums[i] == val:
                nums[i], nums[count - 1] = nums[count - 1], nums[i]
                count -= 1
            else:
                i += 1
        return count
